### ISSUE TITLE
AutoGraph could not transform custom Train_Step function

### ISSUE THREAD

**Issue by YouCantHandleMyOPness** [tensorflow/tensorflow#45922]
**System information**
- Have I written custom code
- OS Platform: Windows 10 64-bit, Desktop
- TensorFlow installed from binary:
- TensorFlow version: 2.3.1
- Python version: 3.7.9
- CUDA/cuDNN version: Cuda compilation tools, release 10.1, V10.1.243
- GPU model and memory: two system linked Nvidia GeForce 2070 RX.  8gb ram each (16gb total).

**Describe the current behavior**
I get the following warning message when using tf.function to convert my function:
"
WARNING:tensorflow:AutoGraph could not transform <function Train_Step at 0x000002935E41C048> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: 'arguments' object has no attribute 'posonlyargs'
"

**Describe the expected behavior**
I expected that when I used a @tf.function decorator the function would compile properly not throw out a warning.

**Standalone code to reproduce the issue**
"
import tensorflow as tf
import numpy as np

batch_size = 100
feature_length = 45
length_target = 1
array_sample_data = np.asarray(np.random.random((batch_size, feature_length)), dtype = np.float32)
array_target_data = np.asarray(np.random.random((batch_size, length_target)), dtype = np.float32)
array_std_data = np.std(array_sample_data, axis = 0)

model = tf.keras.Sequential()
model.add(tf.keras.layers.Dense(50, activation = 'relu', input_shape = [feature_length]))
model.add(tf.keras.layers.Dense(length_target, activation = 'sigmoid'))

optimizer = tf.keras.optimizers.Adam()
loss = tf.keras.losses.MSE
model.compile(optimizer = optimizer, loss = loss)

tensor_sample_data = tf.constant(array_sample_data, dtype = 'float32')
tensor_target_data = tf.constant(array_target_data, dtype = 'float32')
tensor_weight = None

clip_norm = 1
@tf.function	
def Train_Step(tensor_sample_data, tensor_target_data, tensor_weight):
	with tf.GradientTape() as tape:
		y_pred = model.call(tensor_sample_data, training = True)
		loss = model.compiled_loss(y_true = tensor_target_data, y_pred = y_pred, \
sample_weight = tensor_weight, regularization_losses = model.losses) 

	per_replica_gradients = tape.gradient(loss, model.trainable_variables)
	gradients = tf.distribute.get_replica_context().all_reduce('sum', per_replica_gradients) #summed all grads across devices
	loss = tf.distribute.get_replica_context().all_reduce('sum', loss)
	y_pred = tf.distribute.get_replica_context().all_reduce('sum', y_pred)
	(gradients, global_norm_value) = tf.clip_by_global_norm(t_list = gradients, clip_norm = clip_norm)
	model.optimizer.apply_gradients(zip(gradients, model.trainable_variables), experimental_aggregate_gradients = False)

	return (y_pred, loss)

epochs = 3
for index_epoch in range(epochs):
	(y_pred, loss) = Train_Step(tensor_sample_data = tensor_sample_data, tensor_target_data = tensor_target_data, tensor_weight = tensor_weight)
	print('Training loss is {} on epoch {}.'.format(loss, index_epoch + 1))

"
Thanks for your time and consideration.

**Comment by amahendrakar** [comment]
@YouCantHandleMyOPness,
I did not face any warnings while running the code with [TF v2.3](https://colab.research.google.com/gist/amahendrakar/a50c728f9df8d493dfd52831992ce921/45922-2-3.ipynb) and [TF v2.4](https://colab.research.google.com/gist/amahendrakar/f20d1920f1c21cb0e5a488026a6baa7e/45922.ipynb#scrollTo=VRYN13GrRUWF). Please check the linked gist for reference. 

You can safely ignore the warnings and suppress the warning logs by running the below code before importing TensorFlow.  
```
import os
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2' 
import tensorflow as tf
```

Thanks!

**Comment by YouCantHandleMyOPness** [comment]
That's interesting.  I'll just assume that TF 2.3.1 has some strange warning message and that there is nothing to worry about.  Thanks for the help!
